### PR TITLE
0.06

### PR DESCRIPTION
**Pull request by sloppy-genius**
added_firebase_implementation_for_signinactivity

### CODE REVIEW

File: .idea/libraries/android_android_26.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<component name="libraryTable">
```
Comment (pratickRoy) [inline .idea/libraries/android_android_26.xml]: add .idea to .gitignore, uncessary generated files

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 2]
```diff
@@ -14,6 +15,8 @@
 import com.extnds.paparazi.papparazicustomerdroid.BaseFragment;
 import com.extnds.paparazi.papparazicustomerdroid.R;
 import com.extnds.paparazi.papparazicustomerdroid.commons.CommonUtils;
+import com.google.android.gms.tasks.OnCompleteListener;
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: minor: add new line

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 3]
```diff
@@ -89,21 +95,26 @@ private void validateSignInDetails() {
...
-
-        } else {
-
-            Log.d(LOG_TAG, "Invalid Credentials");
-            Toast.makeText(getContext(), "Invalid Credentials, Please Try Again", Toast.LENGTH_SHORT).show();
-        }
-
+        // ToDo : Sync firebase
+        String email = emailEditText.getText().toString().trim();
+        String pass = passwordEditText.getText().toString().trim();
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: minor: use password as name, and make final

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 4]
```diff
@@ -89,21 +95,26 @@ private void validateSignInDetails() {
...
+        firebaseAuth.signInWithEmailAndPassword(email, pass).addOnCompleteListener(this, new OnCompleteListener<AuthResult>()
+        {
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: lambda this?

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 5]
```diff
@@ -89,21 +95,26 @@ private void validateSignInDetails() {
...
+            @Override
+            public void onComplete(@NonNull Task<AuthResult> task)
+            {
+                if(task.isSuccessful())
+                {
+                    CommonUtils.setUserId(getContext(), RandomStringUtils.random(10));
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: random Id?

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java [hunk 6]
```diff
@@ -89,21 +95,26 @@ private void validateSignInDetails() {
...
+                    Toast.makeText(getContext(), "Successfully Logged In", Toast.LENGTH_SHORT).show();
+                    getMainActivity().getSupportFragmentManager().popBackStackImmediate();
+                }
+                else
+                {
+                    Log.d(LOG_TAG, "Invalid Credentials");
```
Comment (pratickRoy) [inline app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: can we specify exactly what went wrong? email or password or both?

Comment (sloppy-genius) [reply app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: Log.d(LOG_TAG, "Invalid Credentials "+task.getException().getMessage());

Comment (sloppy-genius) [reply app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: done

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java (hunk 3 above)
Comment (sloppy-genius) [reply app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: pass -> password and made both password and email final

File: app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java (hunk 2 above)
Comment (sloppy-genius) [reply app/src/main/java/com/extnds/paparazi/papparazicustomerdroid/features/authentication/SignInFragment.java]: done

File: .idea/libraries/android_android_26.xml (hunk 1 above)
Comment (sloppy-genius) [reply .idea/libraries/android_android_26.xml]: added .idea to .gitignore

**Pull request closed without merging**